tests/conn_test.go

```diff
@@ -242,3 +242,33 @@ func TestBlockBufferSize(t *testing.T) {
 	}
 	require.Equal(t, 10000000, i)
 }
+
+func TestReadOnlyUserConnInsert(t *testing.T) {
```

Comment:
We can add other testing scenarios for read-only users:
- should select data from a table
- shouldn't select data from a table 
- shouldn't create a table
- shouldn't delete a table

see https://github.com/ClickHouse/clickhouse-js/blob/01eab4c819992d9f69d94d0d38b952723d1f203c/__tests__/integration/read_only_user.test.ts

Reply:
@gingerwizard do you have other ideas?

Reply:
I've added mentioned scenarios with one-two extra.